src/fact_base.py

```diff
@@ -61,17 +61,27 @@ def shutdown_listener(self, signum, _):
         logging.info(f'Received signal {signum}. Shutting down {self.PROGRAM_NAME}...')
         self.run = False
 
+    def start(self):
+        pass
+
+    def _update_component_workload(self):
+        self.work_load_stat.update()
+
     def shutdown(self):
         logging.info(f'Shutting down components of {self.PROGRAM_NAME}')
         self.work_load_stat.shutdown()
 
     def main(self):
+        self.start()
         logging.info(f'Successfully started {self.PROGRAM_NAME}')
+        counter = 0
         while self.run:
-            self.work_load_stat.update()
+            self._update_component_workload()
             sleep(5)
             if self.args.testing:
                 break
+            if not (counter := counter + 1) % 12:  # only check every minute
```

Comment:
I personally don't like the walrus operator. While I see its usecase in large list comprehensions to improve performance it doesn't improve performance here.
I find this very readable.

Reply:
> I find this very readable.

What is the problem then? :sweat_smile: 
Fun aside, I agree that it's harder to read than without the walrus operator -> changed